### DISCUSSION TITLE
Deploying Ubuntu Server 22.04 fails / Ansible - Unreachable

### DISCUSSION THREAD

**Discussion by javiers91** [vmware/packer-examples-for-vsphere#234] [Help] [answered]
Greetings, I am very very new to this. However I spotted that my build is failing due to ansible being unable to connect to the newly provisioned vm.... which it tries to connect on 127.0.0.1, what I dont know is if its doing so on my local machine OR on the vm its trying to build.
Ansible package is installed on my local machine.

What am I missing here?

```
==> vsphere-iso.linux-<sensitive>: Connected to SSH!
==> vsphere-iso.linux-<sensitive>: Provisioning with Ansible...
    vsphere-iso.linux-<sensitive>: Setting up proxy adapter for Ansible....
==> vsphere-iso.linux-<sensitive>: Executing Ansible: <sensitive>-playbook -e packer_build_name="linux-*****" -e packer_builder_type=vsphere-iso -e packer_http_addr=10.11.100.144:8066 --ssh-extra-args '-o IdentitiesOnly=yes' --extra-vars display_skipped_hosts=false --extra-vars BUILD_USERNAME=***** --extra-vars BUILD_SECRET=***** --extra-vars ANSIBLE_USERNAME=<sensitive> --extra-vars ANSIBLE_SECRET=***** -e <sensitive>_ssh_private_key_file=/tmp/<sensitive>-key2910421335 -i /tmp/packer-provisioner-<sensitive>441382762 /home/jscaglia/pefv/<sensitive>/main.yml
    vsphere-iso.linux-<sensitive>: [DEPRECATION WARNING]: COMMAND_WARNINGS option, the command warnings feature is
    vsphere-iso.linux-<sensitive>:  being removed. This feature will be removed from <sensitive>-core in version 2.14.
    vsphere-iso.linux-<sensitive>:  Deprecation warnings can be disabled by setting deprecation_warnings=False in
    vsphere-iso.linux-<sensitive>: <sensitive>.cfg.
    vsphere-iso.linux-<sensitive>:
    vsphere-iso.linux-<sensitive>: PLAY [all] *********************************************************************
==> vsphere-iso.linux-<sensitive>: failed to handshake
    vsphere-iso.linux-<sensitive>:
    vsphere-iso.linux-<sensitive>: TASK [Gathering Facts] *********************************************************
    vsphere-iso.linux-<sensitive>: fatal: [default]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: Unable to negotiate with 127.0.0.1 port 35939: no matching host key type found. Their offer: ssh-rsa", "unreachable": true}
    vsphere-iso.linux-<sensitive>:
    vsphere-iso.linux-<sensitive>: PLAY RECAP *********************************************************************
    vsphere-iso.linux-<sensitive>: default                    : ok=0    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0
    vsphere-iso.linux-<sensitive>:
==> vsphere-iso.linux-<sensitive>: Provisioning step had errors: Running the cleanup provisioner, if present...
==> vsphere-iso.linux-<sensitive>: Power off VM...
==> vsphere-iso.linux-<sensitive>: Destroying VM...
Build 'vsphere-iso.linux-<sensitive>' errored after 15 minutes 34 seconds: Error executing Ansible: Non-zero exit status: exit status 4

```

**Comment by tenthirtyam** [comment]
Can you please describe the configuration / topology of your environment and which jump host configuration you used - Photon 4, Ubuntu 22.04/22.04, or macOS?

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by javiers91** [reply]
Ok, i think i see where the issue is... I used a fedora 36 VM to run packer. Also ansible --version is failing.
I'll provision a ubuntu 22.04 vm, install all requirements and retest this.

**Reply by tenthirtyam** [reply]
I only test the project against a select number of documented hosts that will run Packer. Your mileage **may** vary with other distros.

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by javiers91** [reply]
Building the Ubuntu vm rn, is there any ansible initial configuration needed besides having the package installed?

**Reply by tenthirtyam** [reply]
Just install via `apt-get` per the README.

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by javiers91** [reply]
actually installing it via apt-get was causing issues (i think), used apt and it worked like a charm
thanks a lot for your help
Now ill tinker with it to remove ovf export to content library

Already left a star, great project!

**Reply by tenthirtyam** [reply]
That's super easy to do in the common vars.

```hcl
// Template and Content Library Settings
common_template_conversion         = true
common_content_library_name        = null
common_content_library_ovf         = false
common_content_library_destroy     = false
common_content_library_skip_export = false

// OVF Export Settings
common_ovf_export_enabled   = false
common_ovf_export_overwrite = true
```

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by javiers91** [reply]
And I went to all the trouble of modifying/commenting the hcl file 😄 
Thanks again Ryan!

**Comment by tenthirtyam** [comment, marked as answer]
Per the README, you may be required to update your `/etc/ssh/ssh_config` or `.ssh/ssh_config` to allow authentication with RSA keys if you are using VMware Photon OS 4.0 or Ubuntu 22.04.

Update to include the following:

```
PubkeyAcceptedAlgorithms ssh-rsa

HostkeyAlgorithms ssh-rsa

```

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Reply by tenthirtyam** [reply]
Please consider adding a GitHub star ⭐️ for the project. It really helps!

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.
